src/common/time/src/timezone.rs

```diff
@@ -15,15 +15,40 @@
 use std::fmt::Display;
 use std::str::FromStr;
 
-use chrono::{FixedOffset, Local, Offset};
+use chrono::FixedOffset;
 use chrono_tz::Tz;
+use once_cell::sync::OnceCell;
 use snafu::{OptionExt, ResultExt};
 
 use crate::error::{
     InvalidTimeZoneOffsetSnafu, ParseOffsetStrSnafu, ParseTimeZoneNameSnafu, Result,
 };
 use crate::util::find_tz_from_env;
 
+/// System time zone in `frontend`/`standalone`,
+/// config by option `default_time_zone` in toml,
+/// default value is `UTC` when `default_time_zone` is not set.
+static DEFAULT_TIME_ZONE: OnceCell<TimeZone> = OnceCell::new();
+
+// Set the System time zone by `tz_str`
+pub fn set_default_time_zone(tz_str: &str) -> Result<()> {
+    let tz = if tz_str.is_empty() {
+        TimeZone::Named(Tz::UTC)
+    } else {
+        TimeZone::from_tz_string(tz_str)?
+    };
+    DEFAULT_TIME_ZONE.get_or_init(|| tz);
+    Ok(())
+}
+
+#[inline(always)]
+/// If the `tz=Some(time_zone)`, return `time_zone` directly,
+/// or return current system time zone.
+pub fn get_time_zone(tz: Option<TimeZone>) -> TimeZone {
+    tz.or(DEFAULT_TIME_ZONE.get().cloned())
+        .unwrap_or(TimeZone::Named(Tz::UTC))
+}
```

Comment:
Make sure `set_default_time_zone` is called and just `unwrap_or(DEFAULT_TIME_ZONE)`. It feels strange to have a default value for `DEFAULT_TIME_ZONE` 😂

Reply:
We can use `get_unchecked` to get `DEFAULT_TIME_ZONE`'s value directly, but it's a `unsafe` operation, may cause unexpected crash, so maybe for robust we can keep `unwrap_or`? After all we have a lot of function depend on `DEFAULT_TIME_ZONE`

Reply:
```suggestion
#[inline(always)]
/// If the `tz=Some(time_zone)`, return `time_zone` directly,
/// or return current system time zone.
pub fn get_time_zone(tz: Option<TimeZone>) -> TimeZone {
    tz.unwrap_or_else(|| {
        DEFAULT_TIME_ZONE
            .get_or_init(|| TimeZone::Named(Tz::UTC))
            .clone()
    })
}
```

In this case, I would suggest this change to prevent trivial compute when `tz` is present.